### DISCUSSION TITLE
Bainite ferritic duplex grain area, aspect ratio and slope statistics

### DISCUSSION THREAD

**Discussion by Mikewangkang** [AzdiarGazder/mtexTools#9] [Q&A] [answered]
Hi， @AzdiarGazder 
Using the algorithm of distinguishing two phases based on GAM and other aspects in mtex, I realized the separation of ferritic and bainite in duplex steel, and also calculated the area and aspect ratio of each grain, and I saw from the paper that the slope of each grain can be calculated using MTEX, but I don't know how to achieve it. For a definition of slope, you can see the illustration attached. And then the area and aspect ratio of each grain are mixed together, and I want to distinguish them according to the phase, that is, the area, aspect ratio and slope of ferrite are one result, and the area, aspect ratio and slope of bainite are one result. I am now manually selecting bainite grain ID with grain area and aspect ratio, which is more time-consuming, can I add code to achieve automatic separation?
![11](https://github.com/AzdiarGazder/mtexTools/assets/139696590/d5ff3954-a9a8-4b0a-86d9-4d21bc1eb9da)
[The following is the paper I referenced.](https://www.mdpi.com/1996-1944/13/19/4236)
Sincerely,
Yours Wangkang.

**Comment by AzdiarGazder** [comment]
Hi @Mikewangkang,

I do not fully understand your question - please let me know if my understanding of your issue is correct.
Are you saying that:
1. you have segmented polygonal ferrite and bainite in a duplex steel using the [ferriteQuantifier](https://github.com/AzdiarGazder/mtexTools/tree/main/ferriteQuantifier) or [kamSegmenter](https://github.com/AzdiarGazder/mtexTools/tree/main/kamSegmenter) algorithms, 
2. you have then saved the new map (in cpr/crc, or ctf or ang or text format), and
3. you now wish to plot the grain aspect ratio for each segmented phase?

If my understanding is correct, then all you have to do is:
- load the saved map in a new script, 
- calculate the grains, and 
- display the aspect ratio.

Here is an example of what I mean.
I assume you have previously:
- run the [ferriteQuantifier](https://github.com/AzdiarGazder/mtexTools/tree/main/ferriteQuantifier) script, and
- saved the segmented map in a file called "newDuplex.ctf"

```matlab
% clear variables
clc; clear all; clear hidden; close all;

% start Mtex
startup_mtex;

% define Mtex plotting convention as X = right, Y = up
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
setMTEXpref('FontSize', 14);

% Import the dataset
ebsd = EBSD.load('newDuplex.ctf','interface','ctf',...
    'convertEuler2SpatialReferenceFrame');

%% Calculate the grains
% identify grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[1 5].*degree,'unitcell');
% remove small clusters
ebsd(grains(grains.grainSize <= 5)) = [];
% re-calculate the grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[1 5].*degree,'unitcell');

% Plot a map of grain aspect ratios for the Polygonal ferrite phase
plot(grains('Polygonal ferrite'),grains('Polygonal ferrite').aspectRatio)
```

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Hope this helps.

Warm regards,
Azdi

**Comment by Mikewangkang** [comment]
Hi, @AzdiarGazder 
Thank you for your reply.I've done the following:
1.I have segmented polygonal ferrite and bainite in a duplex steel using the ferriteQuantifier algorithms,
2.I have then saved the new map (in cpr/crc, or ctf or ang or text format),
3.I've counted the area and the aspect ratioof each grain.
My current question is about the slope of the grains，As I mentioned above, I understand that slope is the angle between the long axis of the grain and the rolling direction of the sample, I saw in an article that MTEX can be counted, but I didn't find the relevant content, the schematic diagram is as follows：
![image](https://github.com/AzdiarGazder/mtexTools/assets/139696590/7b79a336-9c9f-41b5-bda5-4bdb73c2d0b1)
Below is my reference article.
[materials-13-04236.pdf](https://github.com/AzdiarGazder/mtexTools/files/13743543/materials-13-04236.pdf)


Sincerely,
Yours Wangkang.

**Reply by AzdiarGazder** [reply, marked as answer]
Hi @Mikewangkang,

Thank you for the clarification.
I now understand what you're trying to calculate.
Here is the script to compute the angles of the fitted ellipses of all grains.

```matlab
% clear variables
clc; clear all; clear hidden; close all;

% start Mtex
startup_mtex;

% define Mtex plotting convention as X = right, Y = up
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
setMTEXpref('FontSize', 14);

% Import the dataset
ebsd = EBSD.load('newDuplex.ctf','interface','ctf',...
    'convertEuler2SpatialReferenceFrame');

%% Calculate the grains
% identify grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[1 5].*degree,'unitcell');
% remove small clusters
ebsd(grains(grains.grainSize <= 5)) = [];
% re-calculate the grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[1 5].*degree,'unitcell');

%% Calculate the angle and principle axes of the fitted ellipse 
[omega, majorAxis, minorAxis] = principalComponents(grains);

% Plot a grain map of angles of the fitted ellipses for the Polygonal ferrite phase
isPF = grains == grains('Polygonal ferrite');
plot(grains(isPF), omega(isPF)./degree);
```

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Hope this helps.

Warm regards,
Azdi
